packages/styled-components/src/Hooks/useComparisonRoutes.ts

```diff
@@ -0,0 +1,79 @@
+import { useSelector, useDispatch } from 'react-redux';
+import {
+  comparisonReportSelector,
+  hraComparisonReportGetRequest,
+} from '@doctorlink/traversal-redux';
+import { HealthComparisonModel } from '@doctorlink/traversal-core';
+import { useState, useEffect } from 'react';
+
+export type ComparisonRouteName = 'risks' | 'wellbeing' | 'my-numbers';
+
+interface RouteDefinition {
+  path: ComparisonRouteName;
+  enabled: boolean;
+  defer?: boolean;
+}
+
+export interface ComparisonRoutes {
+  routes: ComparisonRouteName[];
+  initialRoute: ComparisonRouteName | undefined;
+  allLoaded: boolean;
+}
+
+export const useComparisonRoutes = (
+  traversalId: string,
+  pastTraversalId: string,
+  riskAtAge: number
+): ComparisonRoutes => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(
+      hraComparisonReportGetRequest(traversalId, pastTraversalId, riskAtAge)
+    );
+  }, [dispatch, traversalId, pastTraversalId, riskAtAge]);
+
+  const comparisonReport = useSelector(
+    comparisonReportSelector
+  ) as HealthComparisonModel;
+  const reportLoaded = comparisonReport.loaded ?? false;
```

Comment:
```suggestion
const comparisonReport = useSelector(comparisonReportSelector);
  const reportLoaded = comparisonReport?.loaded ?? false;
```
Avoid casting and use `?.` instead. (If `HealthAssessmentModel.comparisonReport` is genuinely never null, you probably need to change the type definition.)